wagtail/models/i18n.py

```diff
@@ -174,13 +174,143 @@ def is_active(self) -> bool:
+        # the behavior of the `localized` property on `TranslatableMixin`.
+        if not getattr(settings, "WAGTAIL_I18N_ENABLED", False):
+            return self
+
+        # Get all instances that are available in the active locale. We can find these
+        # by getting all model instances that have a translation key from the original
+        # queryset and that are available in the active locale.
+        active_locale = Locale.get_active()
+        original_translation_keys = self.values_list("translation_key", flat=True)
+        translated_instances = self.model.objects.filter(
+            locale_id=pk(active_locale),
+            translation_key__in=original_translation_keys,
+        )
+
+        # Don't consider draft translations. If a translation is in draft, we want to
+        # use the original instance instead. To do so, we exclude draft translations.
+        # This only applies if the model has a `live` field. We allow bypassing this
+        # behavior by passing `include_draft_translations=True`.
+        from wagtail.models import DraftStateMixin
+
+        if issubclass(self.model, DraftStateMixin) and not include_draft_translations:
+            translated_instances = translated_instances.exclude(live=False)
+
+        # Get all instances that are not available in the active locale. We can find
+        # these by excluding the translation keys for which translations exist from the
+        # original queryset.
+        translated_translation_keys = translated_instances.values_list(
+            "translation_key", flat=True
+        )
+        untranslated_instances = self.exclude(
+            translation_key__in=translated_translation_keys,
+        )
+
+        # Combine the two querysets to get the localized queryset.
+        localized_queryset = self.model.objects.filter(
+            models.Q(pk__in=translated_instances)
+            | models.Q(pk__in=untranslated_instances)
+        )
+
+        if not keep_order:
+            # Apply the same `order_by` as in the original queryset. This does not mean
+            # that the order of the items is retained. Rather, the same fields are used
+            # for ordering. However, the ordering is likely to be different because the
+            # translated values are used.
+            return localized_queryset.order_by(*self.query.order_by)
+        else:
+            # Keep the same order as in the original queryset. To do so, we annotate the
+            # localized queryset with the original order of the translation keys, and
+            # then order by that annotation.
+            ordering_when_clauses = [
+                models.When(translation_key=tk, then=models.Value(index))
+                for index, tk in enumerate(original_translation_keys)
+            ]
+            localized_annotated_queryset = localized_queryset.annotate(
+                original_order=models.Case(*ordering_when_clauses)
+            )
```

Comment:
This is 🤌!

Reply:
Heavily inspired by your `order_by_pk_position` util in Kit 😅